### ISSUE TITLE
Misleading warning "returning a dangling pointer" for output parameters

### ISSUE THREAD

**Issue by mgehre** [mgehre/llvm-project#52]
On
```
bool f(int **p) {
    return true;
}
```
we diagnose `warning: returning a dangling pointer [-Wlifetime]` on the return statement. The warnings is supposed to be about `p`, which is considered an output parameter, but it can be quite confusing. The diagnostic should at least mention `p` and probably avoid the word `returning`.

See https://godbolt.org/z/5HliMv
fyi @Xazax-hun

**Comment by Xazax-hun** [comment]
This makes sense, thanks! I will create a better warning message for that.

**Comment by Xazax-hun** [comment]
Should be fixed in https://github.com/mgehre/llvm-project/commit/5eb7821a95339a776f8fa0f5bca88b72b166311a
